lib/can_do.rb

```diff
@@ -38,18 +40,24 @@ def self.features
     instance.features
   end
 
+  def self.shared_features(name)
+    CONNECTION_POOL.with do |redis|
+      redis.get(FEATURE_KEY_PREFIX+name.to_s)
+    end
+  end
+
   def self.feature?(name, &block)
     name     = name.to_s
     env_name = name.upcase
 
-    fail NotAFeature.new(name) unless features.key?(name)
-
     is_enabled =
-      if ENV.key?(env_name)
+      if !shared_features(name).nil?
```

Comment:
`shared_features(name)` makes a Redis call - so you have 2 in here (not dramatic, but it's good to make this as fast as we can). you can reduce it to 1 by just assigning this stuff to a variable before the conditional.